esmvalcore/_recipe_checks.py

```diff
@@ -220,16 +219,88 @@ def extract_shape(settings):
                 "{}".format(', '.join(f"'{k}'".lower() for k in valid[key])))
 
 
-def valid_multimodel_statistic(statistic):
-    """Check that `statistic` is a valid argument for multimodel stats."""
-    valid_names = ['std'] + list(STATISTIC_MAPPING.keys())
+def _verify_statistics(statistics, step):
+    """Raise error if multi-model statistics cannot be verified."""
+    valid_names = ["mean", "median", "std", "min", "max"]
     valid_patterns = [r"^(p\d{1,2})(\.\d*)?$"]
-    if not (statistic in valid_names
-            or re.match(r'|'.join(valid_patterns), statistic)):
+
+    for statistic in statistics:
+        if not (statistic in valid_names
+                or re.match(r'|'.join(valid_patterns), statistic)):
+            raise RecipeError(
+                "Invalid value encountered for `statistic` in preprocessor "
+                f"{step}. Valid values are {valid_names} "
+                f"or patterns matching {valid_patterns}. Got '{statistic}'.")
+
+
+def _verify_span_value(span):
+    """Raise error if span argument cannot be verified."""
+    valid_names = ('overlap', 'full')
+    if span not in valid_names:
+        raise RecipeError(
+            "Invalid value encountered for `span` in preprocessor "
+            f"`multi_model_statistics`. Valid values are {valid_names}."
+            f"Got {span}.")
+
+
+def _verify_groupby(groupby):
+    """Raise error if groupby arguments cannot be verified."""
+    if not isinstance(groupby, list):
+        raise RecipeError(
+            "Invalid value encountered for `groupby` in preprocessor "
+            "`multi_model_statistics`.`groupby` must be defined as a "
+            f"list. Got {groupby}.")
```

Comment:
wait - you're only checking if it's a list, do the list members get checked anywhere? can I groupby `geese` :duck:

Reply:
You could but it would not do anything because it would not  find any geese attribute. I don't think you can really predict all the possible grouping options so I am not sure how to check it's values.

Reply:
hahah, geese was a red herring of course :fish: Let me ask you this way then - if a user wants to groubpy geese, will there be an exception raised saying geese is not a valid grouping parameter, groupby should be either of [project, dataset, mip ...]? Because you can use that as a restrictor in the check here

Reply:
I don't think we should restrict this to the usual facets (like `project`, `dataset`, etc.) but rather allow arbitrary grouping by e.g. using `groupby: animal` with something like

```yml
- {project: CMIP5, dataset: CanESM2, animal: goose}
- {project: CMIP6, dataset: CanESM5, animal: herring}
- {project: OBS, dataset: ERA5, animal: goose}
- {project: CMIP6 dataset: UKESM}
```

BTW, what happens when at least one dataset does not include the `groupby` attribute like the last example listed here?

Reply:
OK I like that! sounds good to me, better coverage for the mess that'll be CMIP7 :grin: Good point about the bits that don't have the groupby parameter :+1:

Reply:
>BTW, what happens when at least one dataset does not include the groupby attribute like the last example listed here

It would go to another group that has an empty identifier. This case would eventually crash when trying to compute multimodel stats on a single cube.

Reply:
is this explained/pointed to in the docstring/documentation?

Reply:
Added

Reply:
I think this is the only issue left to be addressed. Does this explanation of this behaviour work for you?

Reply:
cheers Saskia, please see my wee bitty code suggestion below. Also, one more (and last, I promise) thing: cheers for tidying up `ensemble_statistics(products, statistics,
                        output_products, keep_input_datasets=True)` docstring - could you please add a quick sentence there saying it's essentially multimodel_statistic but for ensembles - it's not really clear what the func does, juging just from the doctring. Cheers :beer:

Reply:
Yes of course, adding it now.

Reply:
you're a star :star:

Reply:
Is it clearer now?

Reply:
indeed it is, cheers Saskia, this will get a big green check mark from me :smile: